File: API/Entities/Material.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities;

[Table("Material")]
public class Material
{
    public int Id { get; set; }
    public string Name { get; set; }
    public int Price { get; set; }
    public int AmountInStock { get; set; }
    public int Width { get; set; }
    public int Height { get; set; }
    public int Length { get; set; }
    public MaterialType MaterialType { get; set; }
}

File: UI/src/app/models/material.ts
import { MaterialType } from './material-type';

export interface Material {
  id: number;
  name: string;
  price: number;
  amountInStock: number;
  width: number;
  height: number;
  length: number;
  materialTypeName: string;
  materialType: MaterialType;
}
